tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -63,6 +63,55 @@ public ConfigurationBuilder(IConfigurationSource source, IConfigurationTelemetry
 
     public HasKeys WithKeys(string key, string fallbackKey1, string fallbackKey2, string fallbackKey3) => new(_source, _telemetry, key, fallbackKey1, fallbackKey2, fallbackKey3);
 
+    private static bool TryHandleResult<T>(
```

Comment:
nit: It looks like you call this in `GetAs<T>` knowing that the `getDefaultValue` argument is always non-null, and then in `GetAsClass<T> / GetAsStruct<T>` you always pass null. Could you break out the method into two overloads, one where it is present and one where it is not nullable?

Reply:
> Could you break out TryHandleResult<T> into two overloads, one where it is present and one where it is not nullable?

I definitely _could_, I was just trying to reduce the amount of duplication 😅 Maybe it's not worth the hassle 🤷‍♂️

Reply:
So the really _annoying_ thing here is that I can't do that with method overloads. So we're need to call it something like `TryHandleResultWithOptionalDefault` or something 😅  _and_ it still doesn't solve the need to use `!` because of the weirdness with nullables and struct/classes, so I'm not sure it gains a lot overall 🤔